src/battle_animation.cpp

```diff
@@ -203,13 +204,13 @@ void BattleAnimation::UpdateFlashGeneric(int timing_idx, int& r, int& g, int& b,
 void BattleAnimation::UpdateScreenFlash() {
 	int r, g, b, p;
 	UpdateFlashGeneric(screen_flash_timing, r, g, b, p);
-	Main_Data::game_screen->FlashOnce(r, g, b, p, 0);
+	if (!overwrite_screen_effects || r > 0 || g > 0 || b > 0 || p > 0) Main_Data::game_screen->FlashOnce(r, g, b, p, 0);
 }
 
 void BattleAnimation::UpdateTargetFlash() {
 	int r, g, b, p;
 	UpdateFlashGeneric(target_flash_timing, r, g, b, p);
-	FlashTargets(r, g, b, p);
+	if (!overwrite_screen_effects || r > 0 || g > 0 || b > 0 || p > 0) FlashTargets(r, g, b, p);
```

Comment:
I find these entire ``r > 0`` changes (here and two times above) too controversial. Adding another conditional doesn't make this better.
Can we get rid of them for now or move them in another PR?

Reply:
Did some tests and came to the conclusion that these `r > 0` checks are not necessary. The `!overwrite_screen_effects` check is sufficient.